pyingest/parsers/proquest.py

```diff
@@ -1,9 +1,15 @@
-import os
+from __future__ import print_function
+from __future__ import absolute_import
 import re
-import pymarc
+import sys
+if sys.version_info > (3,) or 'unittest' in sys.modules.keys():
+    import pymarc
+else:
+    raise ImportError('Error: ProQuest parser requires an older version of pymarc that is only compatible with Python 2'
+                      ' and it needs to be manually installed. Run the following: pip install "pymarc<=2.9.1"')
```

Comment:
I'm sorry I didn't realize this before. The changes are good but we still have a problem: even after the user has installed `pymarc<=2.9.1`, the code will keep raising this exception. We never import pymarc.

We do not see the problem with the unit tests (that run with python2) because of the condition `or 'unittest' in sys.modules.keys()`.

I propose we import the package and check the version (`pkg_resources` works in python 2 and 3):

```
$ pip install "pymarc<=2.9.1"
$ python
Python 2.7.16 |Anaconda, Inc.| (default, Sep 24 2019, 21:51:30)
[GCC 7.3.0] on linux2
Type "help", "copyright", "credits" or "license" for more information.
>>> import pkg_resources
>>> pkg_resources.get_distribution('pymarc').version
'2.9.1'
```

```
$ pip install -U pymarc
$ python
Python 2.7.16 |Anaconda, Inc.| (default, Sep 24 2019, 21:51:30)
[GCC 7.3.0] on linux2
Type "help", "copyright", "credits" or "license" for more information.
>>> import pkg_resources
>>> pkg_resources.get_distribution('pymarc').version
'3.2.0'
```

So, if it's python 3 or a unit test we just import pymarc without further question. If it is python 2, we use pkg_resources to check the version: we raise the exception if it is not `<=2.9.1`, otherwise we import the package.

Reply:
This might be helpful for string version comparison: https://stackoverflow.com/a/11887885